### ISSUE TITLE
RaceAttack on Android10-12 failed

### ISSUE THREAD

**Issue by fswen16** [SaneBow/PHYjacking-PoC#1]
Hello, I created devices from Android 9 to 12 on Android Studio and launched the RaceAttack successfully on Android 9, but when I launch the RaceAttack on Android 10 to 12, it couldn't start the victim app from the raceattack app but the covering activity was started successfully though it should start after the start of victimapp, so is there something wrong with the code? Or,  I've noticed you have metioned that "upgrading to Android 12 or installing the security patch on Android 8 to 11 (Fig. 11(d)) can mitigate zero-permission fingerprint-jacking attacks", so Android 10(and higher version) on my emulated devices have been pathched which makes me cannot start the victimapp from the raceattack app? Looking forward to your reply, thank you!

**Comment by SaneBow** [comment]
Yes, very likely they are patched. You can check your OS's security patch level. If it's after [2022-05-01](https://source.android.com/docs/security/bulletin/2022-05-01), the vulnerability (CVE-2022-20007) has been patched. Since Android 9 End-of-Life is far before that date, it receives no more security supports and remains vulnerable. To reproduce that attack on Android 10+, you need to install old versions before the patch.
